Marlin/src/pins/stm32f4/pins_CREALITY_CR4NTxxC10.h

```diff
@@ -0,0 +1,258 @@
+
+#define Z_ENABLE_PIN                        PC3
+#define Z_STEP_PIN                          PB6
+#define Z_DIR_PIN                           PB5
+
+#define E0_ENABLE_PIN                       PC3
+#define E0_STEP_PIN                         PB4
+#define E0_DIR_PIN                          PB3
+
+//
+// Temperature Sensors
+//
+#define TEMP_0_PIN                          PC5   // TH1
+#define TEMP_BED_PIN                        PC4   // TB1
+
+//
+// Heaters / Fans
+//
+#define HEATER_0_PIN                        PA1   // HEATER1
+#define HEATER_BED_PIN                      PB10  // HOT BED
+
+#ifndef FAN0_PIN
+  #define FAN0_PIN                          PA0   // FAN
+#endif
+#if PIN_EXISTS(FAN0)
+  #define FAN_SOFT_PWM
+#endif
+
+#if HAS_TMC_UART
+  // Reduce baud rate to improve software serial reliability
+  #define TMC_BAUD_RATE 19200
+
+  /**
+   * TMC2208/TMC2209 stepper drivers
+   *
+   * Hardware serial communication ports.
+   * If undefined software serial is used according to the pins below
+   */
+  #define X_HARDWARE_SERIAL  Serial6
+  #define Y_HARDWARE_SERIAL  Serial6
+  #define Z_HARDWARE_SERIAL  Serial6
+  #define E0_HARDWARE_SERIAL Serial6
+
+  #ifndef X_SLAVE_ADDRESS
+    #define X_SLAVE_ADDRESS 0
+  #endif
+  #ifndef Y_SLAVE_ADDRESS
+    #define Y_SLAVE_ADDRESS 1
+  #endif
+  #ifndef Z_SLAVE_ADDRESS
+    #define Z_SLAVE_ADDRESS 2
+  #endif
+  #ifndef E0_SLAVE_ADDRESS
+    #define E0_SLAVE_ADDRESS 3
+  #endif
+
+  // Software serial
+
+  #define X_SERIAL_TX_PIN                   PC6
+  #define X_SERIAL_RX_PIN                   PC7
+
+  #define Y_SERIAL_TX_PIN        X_SERIAL_TX_PIN
+  #define Y_SERIAL_RX_PIN        X_SERIAL_RX_PIN
+
+  #define Z_SERIAL_TX_PIN        X_SERIAL_TX_PIN
+  #define Z_SERIAL_RX_PIN        X_SERIAL_RX_PIN
+
+  #define E0_SERIAL_TX_PIN       X_SERIAL_TX_PIN
+  #define E0_SERIAL_RX_PIN       X_SERIAL_RX_PIN
+
+  #define E1_SERIAL_TX_PIN       X_SERIAL_TX_PIN
+  #define E1_SERIAL_RX_PIN       X_SERIAL_RX_PIN
+
+  #define X_DIAG_PIN                        PC13
+  #define Y_DIAG_PIN                        PC14
+  #define Z_DIAG_PIN                        PC15
+  #define E0_DIAG_PIN                       PA15
+
+#endif // HAS_TMC_UART
+
+//
+// SD Card
+//
+#define SD_DETECT_PIN                       PC1
+#define SDCARD_CONNECTION                ONBOARD
+#define ONBOARD_SPI_DEVICE                     1
+//#define ONBOARD_SD_CS_PIN                 PA4   // SDSS
+#define SDIO_SUPPORT
+#define NO_SD_HOST_DRIVE                  // This board's SD is only seen by the printer
+
+#if ENABLED(CR10_STOCKDISPLAY) && NONE(RET6_12864_LCD, VET6_12864_LCD)
+  #error "Define RET6_12864_LCD or VET6_12864_LCD to select pins for CR10_STOCKDISPLAY with the Creality V4 controller."
+#endif
+
+#if ENABLED(RET6_12864_LCD)
+
+  // RET6 12864 LCD
+  #define LCD_PINS_RS                       PB12
+  #define LCD_PINS_ENABLE                   PB15
+  #define LCD_PINS_D4                       PB13
```

Comment:
To properly prepare pins files, always define the EXP port(s) on the board. A single EXP port will be "EXP3" and if there are two they are "EXP1" and "EXP2." This will make it easier for us to transition to dedicated LCD pins files, a conversion that we have already begun.